test/e2e/README.md

```diff
@@ -0,0 +1,156 @@
+    PublishPluginBinary(pluginsInfo []\*PluginMeta) (distributionUrls []string, errs []error)
+    // GeneratePluginBundle generates plugin bundle in local file system for given plugin metadata
+    GeneratePluginBundle(pluginsMeta []*PluginMeta) ([]string, []error)
+    // PublishPluginBundle publishes the plugin bundles to given registry bucket and returns the plugins discovery urls
+    PublishPluginBundle(pluginsInfo []*PluginMeta) (discoveryUrls []string, errs []error)
+}
+
+// PluginHelperOps helps to generate and publish plugins
+type PluginHelperOps interface {
+    GeneratePluginOps
+    PublishPluginOps
+}
+```
+
+To perform tanzu plugin command operations:
+
+```go
+// PluginBasicOps helps to perform the plugin command operations
+type PluginBasicOps interface {
+    // ListPlugins lists all plugins by running 'tanzu plugin list' command
+    ListPlugins() ([]PluginListInfo, error)
+}
+
+// PluginSourceOps helps 'plugin source' commands
+type PluginSourceOps interface {
+    // AddPluginDiscoverySource adds plugin discovery source
+    AddPluginDiscoverySource(discoveryOpts \*DiscoveryOptions) error
+}
+
+// PluginCmdOps helps to perform the plugin and its sub-commands lifecycle operations
+type PluginCmdOps interface {
+    PluginBasicOps
+    PluginSourceOps
+}
+```
+
+To perform cluster specific operations:
+
+```go
+// ClusterOps has helper operations to perform on cluster
+type ClusterOps interface {
+    CreateCluster(name string, args []string) (output string, err error)
+    DeleteCluster(name string, args []string) (output string, err error)
+    ClusterStatus(name string, args []string) (output string, err error)
+}
+
+// KindCluster performs k8s KIND cluster operations
+type KindCluster interface {
+    ClusterOps
+}
+```
+
+To perform tanzu config command and CLI lifecycle operations:
+
+```go
+// ConfigLifecycleOps performs "tanzu config" command operations
+type ConfigLifecycleOps interface {
+    ConfigSetFeatureFlag(path, value string) error
+    ConfigGetFeatureFlag(path string) (string, error)
+    ConfigUnsetFeature(path string) error
+    ConfigInit() error
+    ConfigServerList() error
+    ConfigServerDelete(serverName string) error
+    DeleteCLIConfigurationFiles() error
+    IsCLIConfigurationFilesExists() bool
+}
+
+// CliOps performs basic cli operations
+type CliOps interface {
+    CliInit() error
+    CliVersion() (string, error)
+    InstallCLI(version string) error
+    UninstallCLI(version string) error
+}
+```
+
+## Use cases covered in E2E tests
+
+### End user operations
+
+E2E tests are written to validate all CLI Core functionalities from the end-user perspective. They cover all CLI Core commands lifecycle operations. Below is the list of CLI Core commands or use cases covered by the E2E tests:
+
+- CLI lifecycle operations, like build and install the CLI in all possible ways and on all platforms (TODO)
```

Comment:
Could you give a couple of example for "all possible ways"?

Reply:
I mean, installing the CLI with chocolatey, rpm/yum, apt and homebrew.